cmd/osbuild-image-tests/vmwaretest/vmware.go

```diff
@@ -0,0 +1,257 @@
+		return nil, errors.New("GOVMOMI_PASSWORD not set")
+	}
+
+	if !dcExists {
+		return nil, errors.New("GOVMOMI_DATACENTER not set")
+	}
+
+	if !clusterExists {
+		return nil, errors.New("GOVMOMI_CLUSTER not set")
+	}
+
+	if !netExists {
+		return nil, errors.New("GOVMOMI_NETWORK not set")
+	}
+
+	if !dsExists {
+		return nil, errors.New("GOVMOMI_DATASTORE not set")
+	}
+
+	if !folderExists {
+		return nil, errors.New("GOVMOMI_FOLDER not set")
+	}
+
+	return &AuthOptions{
+		Host:       host,
+		Username:   username,
+		Password:   password,
+		Datacenter: datacenter,
+		Cluster:    cluster,
+		Network:    network,
+		Datastore:  datastore,
+		Folder:     folder,
+	}, nil
+}
+
+func ImportImage(creds *AuthOptions, imagePath, imageName string) error {
+	args := []string{
+		"import.vmdk",
+		fmt.Sprintf("-u=%s:%s@%s", creds.Username, creds.Password, creds.Host),
+		"-k=true",
+		fmt.Sprintf("-dc=%s", creds.Datacenter),
+		fmt.Sprintf("-ds=%s", creds.Datastore),
+		imagePath,
+		imageName,
+	}
+	retcode := cli.Run(args)
+
+	if retcode != 0 {
+		return errors.New("importing vmdk failed")
+	}
+	return nil
+}
+
+func DeleteImage(creds *AuthOptions, directoryName string) error {
+	retcode := cli.Run([]string{
+		"datastore.rm",
+		"-f=true",
+		fmt.Sprintf("-u=%s:%s@%s", creds.Username, creds.Password, creds.Host),
+		"-k=true",
+		fmt.Sprintf("-dc=%s", creds.Datacenter),
+		fmt.Sprintf("-ds=%s", creds.Datastore),
+		directoryName + "*", // because vm.create creates another directory with _1 prefix
+	})
+
+	if retcode != 0 {
+		return errors.New("deleting directory failed")
+	}
+	return nil
+}
+
+func WithBootedImage(creds *AuthOptions, imagePath, imageName, publicKey string, f func(address string) error) (retErr error) {
+	vmdkBaseName := filepath.Base(imagePath)
+
+	args := []string{
+		"vm.create",
+		fmt.Sprintf("-u=%s:%s@%s", creds.Username, creds.Password, creds.Host),
+		"-k=true",
+		fmt.Sprintf("-dc=%s", creds.Datacenter),
+		fmt.Sprintf("-ds=%s", creds.Datastore),
+		fmt.Sprintf("-folder=%s", creds.Folder),
+		fmt.Sprintf("-net=%s", creds.Network),
+		"-m=2048", "-g=rhel8_64Guest", "-on=true", "-firmware=bios",
+		fmt.Sprintf("-disk=%s/%s", imageName, vmdkBaseName),
+		"--disk.controller=ide",
+		imageName,
+	}
+	retcode := cli.Run(args)
+	if retcode != 0 {
+		return errors.New("Creating VM from vmdk failed")
+	}
+
+	defer func() {
+		args = []string{
+			"vm.destroy",
+			fmt.Sprintf("-u=%s:%s@%s", creds.Username, creds.Password, creds.Host),
+			"-k=true",
+			imageName,
+		}
+		retcode := cli.Run(args)
+
+		if retcode != 0 {
+			fmt.Printf("Deleting VM %s failed", imageName)
+			return
+		}
+	}()
+
+	// note: by default this will wait/block until an IP address is returned
+	// note: using exec() instead of running the command b/c .Run() returns an int
+	ipAddress, err := exec.Command(
+		"/usr/bin/govc",
+		"vm.ip",
+		fmt.Sprintf("-u=%s:%s@%s", creds.Username, creds.Password, creds.Host),
+		"-k=true",
+		imageName,
+	).Output()
```

Comment:
@teg this is the reason we need `/usr/bin/govc` as a dependency. I wasn't able to figure out how to use `cli.Run()` like I do everywhere else and get the output of the commands instead of an exit code.